### ISSUE TITLE
Referral Source and Referring Doctor pull request

### ISSUE THREAD

**Issue by ghost** [redguava/cliniko-api#88]
Hi,

Could you please add referral source and referring doctor to the pull for patients.

Thank you.
Andrew

**Comment by flyawayfarfar** [comment]
+1

**Comment by hagen** [comment]
Oldie but a good. API endpoint for patients does not include referring Contacts. `patients/<patient_id>/referral_sources` doesn't either. This should be added in.

**Comment by JohnColvin** [comment]
Hey Hagen,

The `patients/#patient_id/referral_source` route should include a Contact referral source. Here's a response from my local dev environment of one:

```json
{
  "links": {
    "self": "http://local.cliniko.dev:3000/v1/patients/510/referral_source"
  },
  "referrer": {
    "links": {
      "self": "http://local.cliniko.dev:3000/v1/contacts/6"
    }
  },
  "referrer_type": "Contact",
  "subcategory": null,
  "notes": null,
  "patient": {
    "links": {
      "self": "http://local.cliniko.dev:3000/v1/patients/510"
    }
  },
  "referral_source_type": {
    "links": {
      "self": "http://local.cliniko.dev:3000/v1/referral_source_types/2"
    }
  }
}
```

The `referrer_type` is `Contact` and the `referrer` links to that contact's endpoint.

Are you getting something different when trying to do this? We won't add this information to the patient's payload. We consider it to be a separate resource.

**Comment by hagen** [comment]
Addressed in #211. Resolved with #212.
@bpinto @JohnColvin How does #211 differ from what was requested here in #88?

**Comment by bpinto** [comment]
I'm not sure I understand, doesn't #212 closes this issue and #211?

**Comment by bpinto** [comment]
Closed by #212
